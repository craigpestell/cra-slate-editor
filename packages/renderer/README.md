# @canner/slate-editor-renderer

[![npm version](https://badge.fury.io/js/%40canner%2Fslate-editor-renderer.svg)](https://badge.fury.io/js/%40canner%2Fslate-editor-renderer)

**NOTE: You should directly use it's plugin instead of writing render function.**

```js
import renderer from "@canner/slate-editor-renderer";
```

Renderers that define corresponding nodes and marks while rendering your editor.

methods:

### commonNode(TagName) => React.Element<TagName>

If you want to render general tag names for nodes such as `p`, `blockquote`, `h1`, `h2`, etc...

### commonMark(TagName, styleKey?, styleValue?)

If you want to render general tag names for nodes such as `span`, `strong`, `code`, `i`, etc...

### emojiNode()

rendering emoji

### imageNode()

rendering video

### linkNode()

renderling inline links

### videoNode(source: 'youtube' | 'dailymotion' | 'youku' | 'vimeo')

rendering video

Supported video sources:

* youtube
* dailymotion
* youku
* vimeo

### Complete example:

```js
// all cases
function renderMark(props) {
  // pass this function to your editor's renderMark prop
  switch (props.mark.type) {
    case "bold":
      return commonMark("strong")(props);
    case "code":
      return commonMark("code")(props);
    case "fontBgColor":
      return commonMark("span", "backgroundColor", "color")(props);
    case "fontColor":
      return commonMark("span", "color", "color")(props);
    case "fontSize":
      return commonMark("span", "fontSize")(props);
    case "letterSpacing":
      return commonMark("span", "letterSpacing")(props);
    case "italic":
      return commonMark("i")(props);
    case "strikethrough":
      return commonMark("s")(props);
    case "underline":
      return commonMark("u")(props);
  }
}

function renderNode(props) {
  // pass this function to your editor's renderNode prop
  switch (props.node.type) {
    case "paragraph":
      return commonNode("p")(props);
    case "blockquote":
      return commonNode("blockquote")(props);
    case "emoji":
      return emojiNode()(props);
    case "heading1":
      return commonNode("h1")(props);
    case "heading2":
      return commonNode("h2")(props);
    case "list-ul":
      return commonNode("ul")(props);
    case "list-ol":
      return commonNode("ol")(props);
    case "list-item":
      return commonNode("li")(props);
    case "image":
      return imageNode()(props);
    case "link":
      return linkNode()(props);
    case "youtube":
      return videoNode("youtube")(props);
    case "dailymotion":
      return videoNode("dailymotion")(props);
    case "youku":
      return videoNode("youku")(props);
    case "vimeo":
      return videoNode("vimeo")(props);
  }
}
```

# @canner/slate-editor-html

Serializer and deserializer
