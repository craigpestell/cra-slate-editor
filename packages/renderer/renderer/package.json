{
  "name": "@canner/slate-editor-renderer",
  "version": "1.5.0",
  "description": "Renderers for slate nodes and marks",
  "main": "lib/index.js",
  "author": "",
  "license": "MIT",
  "publishConfig": {
    "access": "public"
  },
  "peerDependencies": {
    "react": "^0.14.0 || ^15.0.0 || 16.x",
    "react-dom": "^0.14.0 || ^15.0.0 || 16.x",
    "slate": "^0.33.0",
    "styled-components": "^3.2.1"
  },
  "dependencies": {
    "@canner/slate-helper-inline-adddata": "^1.5.0",
    "emoji-mart": "^2.5.0",
    "js-video-url-parser": "^0.2.7",
    "lodash.mapvalues": "^4.6.0",
    "prismjs": "^1.15.0",
    "quill-icons": "^1.5.0",
    "react-icons": "^2.2.7",
    "react-loading-image": "^0.5.0",
    "react-resizable": "^1.7.5",
    "slate-edit-table": "^0.15.0-alpha.4"
  },
  "gitHead": "bb30f8746be7cbe2cda641b6b74c9d976bb56e31"
}
