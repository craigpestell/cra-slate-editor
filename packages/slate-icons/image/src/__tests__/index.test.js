import React from "react";
import { Change, Value } from "slate";
import Image from "../";
import renderer from "react-test-renderer";

test("create a Image icon", () => {
  const initialValue = Value.fromJSON({
    document: {
      nodes: [
        {
          object: "block",
          type: "paragraph",
          nodes: [
            {
              object: "text",
              leaves: [
                {
                  text: "A line of text in a paragraph."
                }
              ]
            }
          ]
        }
      ]
    }
  });

  const component = renderer.create(
    <Image change={new Change({ value: initialValue })} />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
