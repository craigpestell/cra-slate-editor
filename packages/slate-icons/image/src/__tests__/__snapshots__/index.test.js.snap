// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create a Image icon 1`] = `
<div
  style={
    Object {
      "display": "inline-block",
    }
  }
>
  <span
    data-active={false}
    onClick={[Function]}
    onMouseDown={[Function]}
  >
    <svg
      className={undefined}
      height="18px"
      onChange={undefined}
      viewBox="0 0 18 18"
      width="18px"
    >
      <rect
        className="ql-stroke"
        height="10"
        width="12"
        x="3"
        y="4"
      />
      <circle
        className="ql-fill"
        cx="6"
        cy="7"
        r="1"
      />
      <polyline
        className="ql-even ql-fill"
        points="5 12 5 11 7 9 8 10 11 7 13 9 13 12 5 12"
      />
    </svg>
  </span>
</div>
`;
